third_party/nvidia/backend/driver.py

```diff
@@ -212,6 +212,15 @@ def format_of(ty):
 static void _launch(int gridX, int gridY, int gridZ, int num_warps, int num_ctas, int clusterDimX, int clusterDimY, int clusterDimZ, int shared_memory, CUstream stream, CUfunction function, CUdeviceptr global_scratch{', ' + arg_decls if len(arg_decls) > 0 else ''}) {{
   void *params[] = {{ {', '.join(params)} }};
   if (gridX*gridY*gridZ > 0) {{
+    CUcontext pctx;
+    CUDA_CHECK(cuCtxGetCurrent(&pctx));
```

Comment:
How long does it take to get a context? Would it add significant latency?

Reply:
Fair question.  I ran the launch latency benchmark I wrote a few months back (now part of [TritonBench](https://github.com/pytorch-labs/tritonbench)), and it looks like _maybe_ it adds 0.1us (but that's probably below the noise threshold of my box):

Before:
```
$ python run.py --op launch_latency --only nop_triton_compiled_kernel_run
  x_val    nop_triton_compiled_kernel_run-walltime
-------  -----------------------------------------
      0                                 0.00292743
     19                                 0.00431097
```

After:
```
% python run.py --op launch_latency --only nop_triton_compiled_kernel_run
  x_val    nop_triton_compiled_kernel_run-walltime
-------  -----------------------------------------
      0                                 0.00302734
     19                                 0.00438724
```